src/Symfony/Component/VarDumper/Server/DumpServer.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Server;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\VarDumper\Cloner\Data;
+
+/**
+ * A server collecting Data clones sent by a ServerDumper.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @final
+ */
+class DumpServer
+{
+    public const HOST_ENV_VAR = 'VAR_DUMPER_SERVER';
+    public const DEFAULT_HOST = 'tcp://0.0.0.0:9912';
+
+    private $host;
+    private $socket;
+    private $logger;
+
+    /**
+     * @param string|null          $host   The server host or null to read it from the VAR_DUMPER_SERVER env var
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct(string $host = null, LoggerInterface $logger = null)
+    {
+        if (null === $host) {
+            $host = getenv(static::HOST_ENV_VAR) ?: static::DEFAULT_HOST;
```

Comment:
not sure we need handling of env here - can't we use DI based env injection instead?

Reply:
As discussed in https://github.com/symfony/symfony/pull/23831#discussion_r132812863 & https://github.com/symfony/symfony/pull/23831#discussion_r132816454, I'd like to keep it for using it easily from almost anywhere, even without the debug bundle and for multiple projects at the same time.